hydroflow/tests/surface_join_fused.rs

```diff
@@ -0,0 +1,211 @@
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::rc::Rc;
+
+use hydroflow::lattices::set_union::SetUnionSingletonSet;
+use hydroflow::{assert_graphvis_snapshots, hydroflow_syntax};
+use lattices::set_union::SetUnionHashSet;
+use lattices::Merge;
+use multiplatform_test::multiplatform_test;
+
+macro_rules! assert_contains_each_by_tick {
+    ($results:expr, $tick:expr, $input:expr) => {{
+        for v in $input {
+            assert!(
+                $results.borrow()[&$tick].contains(v),
+                "did not contain: {:?} in {:?}",
+                v,
+                $results.borrow()[&$tick]
+            );
+        }
+    }};
+}
```

Comment:
Would be nice to get something like this for the assert op

Reply:
made an issue for it.